File: backend/ResearchCruiseApp/Application/Models/DTOs/Forms/FormAInitValuesDto.cs
using Newtonsoft.Json;
using Newtonsoft.Json.Serialization;
using ResearchCruiseApp.Application.Models.DTOs.CruiseApplications;

namespace ResearchCruiseApp.Application.Models.DTOs.Forms;


[JsonObject(NamingStrategyType = typeof (CamelCaseNamingStrategy))]
public class FormAInitValuesDto
{
    public List<FormUserDto> CruiseManagers { get; set; } = [];
    
    public List<FormUserDto> DeputyManagers { get; set; } = [];
    
    public List<string> Years { get; set; } = [];
    
    public List<string> ShipUsages { get; set; } = [];
    
    public List<ResearchAreaDto> ResearchAreas { get; set; } = [];
    
    public List<string> CruiseGoals { get; set; } = [];
    
    public List<ResearchTaskDto> HistoricalResearchTasks { get; set; } = [];
    
    public List<ContractDto> HistoricalContracts { get; set; } = [];
    
    public List<UgUnitDto> UgUnits { get; set; } = [];
    
    public List<string> HistoricalGuestInstitutions { get; set; } = [];
    
    public List<SpubTaskDto> HistoricalSpubTasks { get; set; } = [];
    public List<PublicationDto> HistoricalPublications { get; set; } = [];
}

File: frontend/src/modules/cruise-applications/models/FormAInitValuesDto.ts
import { ContractDto } from '@/cruise-applications/models/ContractDto';
import { FormUserDto } from '@/cruise-applications/models/FormUserDto';
import { PublicationDto } from '@/cruise-applications/models/PublicationDto';
import { ResearchAreaDto } from '@/cruise-applications/models/ResearchAreaDto';
import { ResearchTaskDto } from '@/cruise-applications/models/ResearchTaskDto';
import { SpubTaskDto } from '@/cruise-applications/models/SpubTaskDto';
import { UgUnitDto } from '@/cruise-applications/models/UgUnitDto';

export type FormAInitValuesDto = {
  cruiseManagers: FormUserDto[];
  deputyManagers: FormUserDto[];
  years: string[];
  shipUsages: string[];
  researchAreas: ResearchAreaDto[];
  cruiseGoals: string[];
  historicalResearchTasks: ResearchTaskDto[];
  historicalContracts: ContractDto[];
  ugUnits: UgUnitDto[];
  historicalGuestInstitutions: string[];
  historicalSpubTasks: SpubTaskDto[];
  historicalPublications: PublicationDto[];
};
